src/session.clj

```diff
@@ -0,0 +1,56 @@
+(ns session
+  (:gen-class)
+    (require [session :refer :all])
+    (require [display :refer :all])
+    (require [board :refer :all]))
+
+(defn session-welcome
+  []
+  "Welcome to Tic-Tac-Toe! You will be 'X' and the computer will be 'O'\nThe spaces available are labeled 1 through 9.")
+
+(defn numeric? [s]
+  (if-let [s (seq s)]
+    (let [s (if (= (first s) \-) (next s) s)
+          s (drop-while #(Character/isDigit %) s)
+          s (if (= (first s) \.) (next s) s)
+          s (drop-while #(Character/isDigit %) s)]
+      (empty? s))))
+
+(defn parse-int [s]
+   (Integer. (re-find  #"\d+" s )))
+
+(defn session-prompt
+  [msg]
+  (display-text msg)
+  (let [input (read-line)]
```

Comment:
I would wrap the read-line into a function similar to how you wrapped println into display-text.

Reply:
Ok. I have changed it to this:   
  (let [input (String->Number (read-line))]
  (if (validate-entry input)
    (- input 1)
    (recur "Invalid Input. Please enter a valid move: "))))

Should I combine reading input + validating into one function?

Reply:
No I wouldn't combine them. you should wrap read-line into a function like prompt-input similar to what you did with display-text. That way you are coupled to read-line and can change the implementation later if needed.